pkg/client/daemon/tunrouter.go

```diff
@@ -210,7 +211,13 @@ func (t *tunRouter) setOutboundInfo(ctx context.Context, mi *daemon.OutboundInfo
 		}
 
 		dgroup.ParentGroup(ctx).Go("watch-cluster-info", func(ctx context.Context) error {
-			return t.watchClusterInfo(ctx, kubeDNS)
+			err := t.watchClusterInfo(ctx, kubeDNS)
+			if errors.Is(err, &client.RecvEOF{}) {
```

Comment:
`errors.Is` is for error values; `errors.As` is for error types.

So for example, you'd use `errors.Is(err, os.ErrExist)` but `errors.As(err, &os.PathError{})`.

So I'm thinking that this should be `errors.As`.